File: solutions/typescript/VB_T4_003_even_odd_mutual_recursion.ts
// VB-T4-003: Even/Odd -- TypeScript baseline
function isEven(n: number): boolean { return n === 0 ? true : isOdd(n - 1); }
function isOdd(n: number): boolean { return n === 0 ? false : isEven(n - 1); }
console.assert(isEven(4) === true);
console.assert(isEven(7) === false);

File: solutions/python/VB_T4_003_even_odd_mutual_recursion.py
"""VB-T4-003: Even/Odd Mutual Recursion -- Python baseline."""
def is_even(n: int) -> bool:
    if n == 0: return True
    return is_odd(n - 1)

def is_odd(n: int) -> bool:
    if n == 0: return False
    return is_even(n - 1)

if __name__ == "__main__":
    assert is_even(4) is True
    assert is_even(7) is False
    print("VB-T4-003 ok")
